src/main/java/com/iexec/resultproxy/jwt/JwtService.java

```diff
@@ -80,13 +80,24 @@ private byte[] initKey(String jwtKeyPath) throws IOException {
         return Base64.getDecoder().decode(payload);
     }
 
+    /**
+     * Retrieves existing JWT from the database or creates it.
+     * <p>
+     * A new signed token will be created in the following cases:
+     * <ul>
+     * <li> If a JWT is not found, an exception will be thrown and a new token will be created.
+     * <li> If a JWT is found but was not signed with the correct key.
+     * @param walletAddress Ethereum address for which
+     * @return A valid JWT token signed with this instance key.
+     */
     private String getOrCreateJwt(String walletAddress) {
         String jwtString;
-        Optional<Jwt> oExistingJwt = findByWalletAddress(walletAddress);
-
-        if (oExistingJwt.isPresent()) {
-            jwtString = oExistingJwt.get().getJwtString(); // TODO generate new token
-        } else {
+        try {
+            Jwt jwt = findByWalletAddress(walletAddress).orElseThrow();
+            jwtString = jwt.getJwtString();
+            getWalletAddressFromJwtString(jwtString);
+        } catch (RuntimeException e) {
```

Comment:
I'm not sure we want to catch every `RuntimeException`. Isn't it too wide?
```suggestion
        } catch (JwtException | NoSuchElementException e) {
```

Reply:
I agree with Maxence (https://github.com/jwtk/jjwt#quickstart).